common/lib/capa/capa/capa_problem.py

```diff
@@ -191,6 +193,48 @@ def __init__(self, problem_text, id, capa_system, state=None, seed=None):
 
         self.extracted_tree = self._extract_html(self.tree)
 
+    def make_xml_compatible(self, tree):
+        """
+        Adjust tree xml in-place for compatibility, such as
+        supporting an old xml format by translating it to a new format.
```

Comment:
At the very least I'd want this docstring to be assertive in saying what's going on, eg, "Translates old-style XML to new-style XML. New style XML provides HTML attributes in the XML definition of problems that indicate when an option is or isn't correct. This enhances accessibility" (or something like that)